python/summarize_sgfs.py

```diff
@@ -0,0 +1,295 @@
+    self.results = {}  # a dictionary as {(black_player_name, white_player_name):Record}
+    self._all_sgfs_files = set()
+    self._all_sgf_files = set()
+    self._new_sgfs_files = set()
+    self._new_sgf_files = set()
+    self._warning = False
+
+  def print_elos(self) -> elo.EloInfo:
+    """Print estimated Bayes Elo scores centered as Elo prior"""
+    elo_info = self._estimate_elo()
+    print(elo_info)
+    print(f"The prior is set as {self._elo_prior}, and the estimated Bayes Elo:")
+    if self._warning:
+      warnings.warn("There are handicap games or games with komi < 5.5 or komi > 7.5")
+    return elo_info
+
+  def print_game_results(self):
+    """Print game results in a matrix with item in row i and column j as the numbers of games player i beaten player
+    j. We split draw game as 0.5 to each player."""
+    print("Player information:")
+    self._print_player_info()
+    print("Game Results by Player ID:")
+    self._print_result_matrix()
+    return
+
+  def get_game_results(self) -> Dict:
+    """Return a dictionary of game results as {(black_player_name, white_player_name):Record}
+      We can retrieve results by player's name as
+      results[(black_player_name, white_player_name)].win
+      results[(black_player_name, white_player_name)].lost
+      results[(black_player_name, white_player_name)].draw
+    """
+    return self.results
+
+  """ Private functions starts here, and no more public functions below """
+
+  def _add_game_file_names(self, input_file_dir, search_subdir):
+    if not os.path.exists(input_file_dir):
+      print(f"There is no directory under name {input_file_dir}")
+      return None
+    _files = []
+    if (search_subdir):
+      for (dirpath, dirnames, filenames) in os.walk(input_file_dir):
+        _files += [os.path.join(dirpath, file) for file in filenames]
+    else:
+      _files = [os.path.join(input_file_dir, file) for file in os.listdir(input_file_dir)]
+
+    _new_sgfs_files = set([file for file in _files if file.split(".")[-1].lower() ==
+                           "sgfs"])
+    _new_sgf_files = set([file for file in _files if file.split(".")[-1].lower() == "sgf"])
+
+    self._new_sgfs_files = _new_sgfs_files.difference(self._all_sgfs_files)
+    self._new_sgf_files = _new_sgf_files.difference(self._all_sgf_files)
+
+    self._all_sgfs_files = self._all_sgfs_files.union(_new_sgfs_files)
+    self._all_sgf_files = self._all_sgf_files.union(_new_sgf_files)
+    print(f"We found {len(self._new_sgfs_files)} new sgfs files and {len(self._new_sgf_files)} new sgf files in the "
+          f"search "
+          f"directory {input_file_dir}.")
+
+  def _add_a_game_file_name(self, input_file_name):
+    if not os.path.isfile(input_file_name):
+      print(f"There is no file with name {input_file_name}.")
+      return None
+    if input_file_name.split(".")[-1].lower() == "sgf":
+      self._new_sgf_files = {input_file_name}.difference(self._all_sgf_files)
```

Comment:
Once you no longer have _new_sgf_files field, you can replace this line and the following line with:

```if input_file_name in self._all_sgf_files:```

and similar for other places where you test membership of just one file.

Reply:
Agreed and changed.